pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -178,30 +178,25 @@ def save_checkpoint(self, trainer, pl_module):
         """
         Performs the main logic around saving a checkpoint
         """
-        # only run on main process
-        if trainer.global_rank != 0:
-            return
+        epoch = trainer.current_epoch
+        # track epoch when ckpt was last checked
+        self.epoch_last_check = trainer.current_epoch
 
-        # no models are saved
-        if self.save_top_k == 0:
-            return
 
```

Comment:
```suggestion
```

Reply:
Fixed some things. Do you still want me to remove this line?

Reply:
all good, it was pep8 issue before.